resolver/src/hosts.rs

```diff
@@ -89,21 +99,19 @@ pub fn read_hosts_conf<P: AsRef<Path>>(path: P) -> io::Result<Hosts> {
 
 /// parse &str to RData::A or RData::AAAA
 pub fn parse_literal_ip(addr: &str) -> Option<RData> {
-    if let Ok(ip4) = addr.parse::<Ipv4Addr>() {
-        return Some(RData::A(ip4));
-    }
-    if let Ok(ip6) = addr.parse::<Ipv6Addr>() {
-        return Some(RData::AAAA(ip6));
+    match IpAddr::from_str(addr) {
+        Ok(IpAddr::V4(ip4)) => Some(RData::A(ip4)),
+        Ok(IpAddr::V6(ip6)) => Some(RData::AAAA(ip6)),
+        Err(_) => None,
```

Comment:
Can you add a `warn!("could not parse an IP from hosts file: {}", e)` here? But still return None?

That way we're not just dropping this on the floor...